collect_app/src/main/java/org/odk/collect/android/widgets/UrlWidget.java

```diff
@@ -17,56 +17,65 @@
 import android.annotation.SuppressLint;
 import android.content.Context;
 import android.net.Uri;
+import android.util.TypedValue;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
 import android.widget.Button;
-import android.widget.LinearLayout;
 import android.widget.TextView;
 import android.widget.Toast;
 
 import org.javarosa.core.model.data.IAnswerData;
 import org.javarosa.core.model.data.StringData;
+import org.javarosa.form.api.FormEntryPrompt;
 import org.odk.collect.android.R;
 import org.odk.collect.android.formentry.questions.QuestionDetails;
-import org.odk.collect.android.formentry.questions.WidgetViewUtils;
 import org.odk.collect.android.utilities.CustomTabHelper;
-import org.odk.collect.android.widgets.interfaces.ButtonWidget;
-
-import static org.odk.collect.android.formentry.questions.WidgetViewUtils.createSimpleButton;
-import static org.odk.collect.android.formentry.questions.WidgetViewUtils.getCenteredAnswerTextView;
+import org.odk.collect.android.utilities.MultiClickGuard;
 
 /**
  * Widget that allows user to open URLs from within the form
  *
  * @author Yaw Anokwa (yanokwa@gmail.com)
  */
 @SuppressLint("ViewConstructor")
-public class UrlWidget extends QuestionWidget implements ButtonWidget {
+public class UrlWidget extends QuestionWidget {
 
-    private Uri uri;
-    final Button openUrlButton;
-    final TextView stringAnswer;
     private final CustomTabHelper customTabHelper;
+    private Uri uri;
 
-    public UrlWidget(Context context, QuestionDetails questionDetails) {
+    protected Button openUrlButton;
+    protected TextView stringAnswer;
+
+    public UrlWidget(Context context, QuestionDetails questionDetails, CustomTabHelper customTabHelper) {
         super(context, questionDetails);
+        this.customTabHelper = customTabHelper;
+    }
+
+    @Override
+    protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
+        ViewGroup answerView = (ViewGroup) LayoutInflater.from(context).inflate(R.layout.url_widget_answer, null);
 
-        openUrlButton = createSimpleButton(getContext(), getFormEntryPrompt().isReadOnly(), context.getString(R.string.open_url), getAnswerFontSize(), this);
+        openUrlButton = answerView.findViewById(R.id.url_button);
+        stringAnswer = answerView.findViewById(R.id.url_answer_text);
 
-        stringAnswer = getCenteredAnswerTextView(getContext(), getAnswerFontSize());
+        if (prompt.isReadOnly()) {
+            openUrlButton.setVisibility(GONE);
+        } else {
+            openUrlButton.setTextSize(TypedValue.COMPLEX_UNIT_DIP, answerFontSize);
+            openUrlButton.setOnClickListener(v -> {
```

Comment:
You can simplify this: `openUrlButton.setOnClickListener(v ->  onButtonClick());`